ansible-galaxy - support ``resolvelib >= 0.5.3, < 0.10.0``

hswong3i opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
https://pypi.org/project/resolvelib/0.9.0/ released on 2022-11-17:

  * https://github.com/sarugaku/resolvelib/blob/master/CHANGELOG.rst#090-2022-11-17
  * https://github.com/sarugaku/resolvelib/releases/tag/0.9.0

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

s-hertel reviewed (approved):
Looks good, can you add a minor_changes changelog fragment?  https://docs.ansible.com/ansible/latest/community/development_process.html#creating-changelog-fragments

hswong3i commented on changelogs/fragments/79399-resolvelib_lt_0_10_0.yml:
```diff
@@ -0,0 +1,2 @@
+major_changes:
+  - Upgrade `resolvelib >= 0.5.3, < 0.10.0`
```
@s-hertel does this good enough?

GhostLyrics replied:
Didn't the reviewer request "minor" changes, where you used "major"?

hswong3i replied:
Sorry my miss, typo mistake, fixed and force pushed.

mattclay commented on test/lib/ansible_test/_data/requirements/ansible.txt:
```diff
@@ -12,4 +12,4 @@ packaging
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.9.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 0.10.0  # dependency resolver used by ansible-galaxy
```
Since `resolvelib` is pre-installed in our test containers, we should temporarily pin the latest version allowed to force the tests to run with that version. Once the tests have passed that commit can be reverted.

Please make those changes without force pushing, so that we keep the reference to the passing CI run. We can squash the commits when merging the PR so the temporary commits won't be in the final commit merged to the `devel` branch.


```suggestion
resolvelib == 0.9.0  # dependency resolver used by ansible-galaxy
```

hswong3i replied:
https://github.com/ansible/ansible/pull/79399/commits/0d3544bacab2935a90827dfd25fa7cf7e9613899 already pass CI.

@mattclay your suggested change also commit as additional for trigger CI, too.

hswong3i committed:
Upgrade `resolvelib >= 0.5.3, < 0.10.0`

https://pypi.org/project/resolvelib/0.9.0/ released on 2022-11-17:

  * https://github.com/sarugaku/resolvelib/blob/master/CHANGELOG.rst#090-2022-11-17
  * https://github.com/sarugaku/resolvelib/releases/tag/0.9.0

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

hswong3i committed:
Trigger CI by pinning resolvelib with latest version

Since resolvelib is pre-installed in our test containers, we should temporarily pin the latest version allowed to force the tests to run with that version. Once the tests have passed that commit can be reverted.

Please make those changes without force pushing, so that we keep the reference to the passing CI run. We can squash the commits when merging the PR so the temporary commits won't be in the final commit merged to the devel branch.

Co-authored-by: Matt Clay <matt@mystile.com>

hswong3i commented:
@mattclay https://github.com/ansible/ansible/pull/79399/commits/5bf1fcc3dd76a80c14962c1786c54da2f3c1889f passed CI, so we may squash it before merge https://github.com/ansible/ansible/pull/79399/commits/0d3544bacab2935a90827dfd25fa7cf7e9613899

mattclay reviewed (commented):
Tests passed with 0.9.0. @hswong3i You can push another commit now that reverts the temporary change. We'll squash all 3 commits when merging the PR.

hswong3i committed:
Revert "Trigger CI by pinning resolvelib with latest version
"

This reverts commit 5bf1fcc3dd76a80c14962c1786c54da2f3c1889f.

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

hswong3i commented:
@mattclay any else follow up required from my side? Please kindly advice ;-)

s-hertel commented on changelogs/fragments/79399-resolvelib_lt_0_10_0.yml:
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - Upgrade `resolvelib >= 0.5.3, < 0.10.0`
```
```suggestion
  - ansible-galaxy - support ``resolvelib >= 0.5.3, < 0.10.0``.
```

hswong3i replied:
Let's do it ;-)

hswong3i committed:
Update changelogs/fragments/79399-resolvelib_lt_0_10_0.yml

Co-authored-by: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>

s-hertel commented:
@hswong3i Thanks for working on this!

s-hertel commented:
Features aren't normally backported. Seems low risk, but @sivel / @nitzmahone would need to make an exception.

gotmax23 commented:
Ack. I don't need this backported; I just wanted to check so I could update the Fedora resolvelib package if necessary.

hswong3i commented:
I had backport it for 2.11 since 7 days ago and working perfectly:

https://github.com/alvistack/ansible-ansible/commit/c6a691856500d8c8ce891eaacedf70ba375ffff1

IMHO we could and should backport this for previous stable release, too?

gotmax23 commented:
It should be safe given that the lower version constraint remains the same.

hswong3i commented:
Backport as below:
* https://github.com/ansible/ansible/pull/79458
* https://github.com/ansible/ansible/pull/79457

s-hertel commented:
@hswong3i As mentioned, not sure if this is backportable (nothing to do with whether it works). But if it is, you'd need to open a backport for 2.14 as well - we shouldn't skip a version.
